src/main/java/pl/edu/pw/mini/ingreedio/api/controller/ProductController.java

```diff
@@ -29,28 +35,41 @@
 @Tag(name = "Products" /*, description = "..."*/)
 public class ProductController {
     private final ProductService productService;
+    private final ProductDtoMapper productDtoMapper;
+    private final int pageSize = 10;
 
     @Operation(summary = "Get matching products", description = "Get matching products")
     @GetMapping
-    public ResponseEntity<List<ProductDto>> getProducts(
+    public ResponseEntity<ProductListResponseDto> getProducts(
```

Comment:
I would create `ProductListRequestDto` and `getPaginatedProducts(ProductListRequest request)` in the `ProductService`, then move the logic from that controller to the service function.